lib/constructs/interface.js

```diff
@@ -172,6 +219,13 @@ Interface.prototype.generateExport = function () {
     this.setup(obj, constructorArgs, privateData);
     return utils.implForWrapper(obj);
   },
+  createDefaultIterator(target, kind) {
+    const iterator = Object.create(IteratorPrototype);
```

Comment:
The spec just says that the iterator object must have the prototype object as `[[Prototype]]`, but it does not say how the object is to be constructed. It doesn't say anything about `constructor` property, so I guess we could make a wrapper function in the form of

```js
function Iterator(target, kind) {
  this[utils.iterTargetSymbol] = target;
  this[utils.iterKindSymbol] = kind;
  this[utils.iterIndexSymbol] = 0;
}
Iterator.prototype = Object.create(IteratorPrototype);
// note `Iterator.prototype.constructor` is not set
```

That should allow us to get rid of the `createDefaultIterator` also, though if that would be desirable is another question.

The main question is why not `Object.create`?